src/widget/wtracktext.cpp

```diff
@@ -56,12 +56,18 @@ void WTrackText::dragEnterEvent(QDragEnterEvent *event) {
 
 void WTrackText::dropEvent(QDropEvent *event) {
     if (DragAndDropHelper::allowLoadToPlayer(m_pGroup, m_pConfig)) {
```

Comment:
Correct me if I'm wrong, but it looks like the exact same code is duplicated in several widgets. Somehow this should be centralized to one location. How about making a simple TrackDropTarget class that only implements a dropEvent method and making all these widgets inherit from it?

Reply:
I made a `WTrackDropTarget` class and tried to make stuff inherit from it, but it causes there to be multiple `QWidget` classes in the inheritance tree. I haven't tried it with virtual inheritance yet which might fix that, but it might also be easier just to implement a `DragAndDropHelper::trackDropEvent()` function and just call that from  `dropEvent()` (and a similar helper for `dragEnterEvent()`).

Reply:
Okay, as long as the code isn't duplicated. :+1: